protocols/grpc/agent.proto

```diff
@@ -78,6 +78,12 @@ message CreateContainerRequest {
 	// The agent would receive an OCI spec with PID namespace cleared
 	// out altogether and not just the pid ns path.
 	bool sandbox_pidns = 7;
+
+	// This field is used to indicate if container should just join the
+	// the pid namespace of the agent. This functionality is added to
+	// allow debug containers/sidecars to have access to the main pid
+	// namespace.
+	bool agent_pidns = 8;
```

Comment:
Idea: given this is a per container setting and given there are potential security implications of allowing anything to run in the agents NS, we might want to consider passing a similar bool to the agent on startup that says whether the agent should honour this setting. By default, it would either ignore it (and log that fact), or error. However, if `agent_pidns=true` is set in `configuration.toml`, it would be honoured. In fact, if this is *only* to be used in debug scenarios, maybe we should also only honour this setting if `enable_debug=true` for the agent too?

Reply:
Agree, it makes sense to me to add an additional configuration  called `agent_pidns` in the configuration file to have this feature enabled. That way only the only the root user can enable this feature. I am not so sure about adding the requirement for enable_debug flag as well, will give this some thought.